store/src/main/java/org/apache/rocketmq/store/ha/HAService.java

```diff
@@ -280,9 +280,14 @@ private void doWaitTransfer() {
                 if (!this.requestsRead.isEmpty()) {
                     for (CommitLog.GroupCommitRequest req : this.requestsRead) {
                         boolean transferOK = HAService.this.push2SlaveMaxOffset.get() >= req.getNextOffset();
-                        for (int i = 0; !transferOK && i < 5; i++) {
+                        long waitUntillWhen = HAService.this.defaultMessageStore.getSystemClock().now()
+                            + HAService.this.defaultMessageStore.getMessageStoreConfig().getSyncFlushTimeout();
+                        while (HAService.this.defaultMessageStore.getSystemClock().now() < waitUntillWhen) {
```

Comment:
you should check transferok to reduce unnecessary block。

Reply:
Yes,  transferOk was checked first(line282) and then enter while loop. In the while loop, it would check transferOk again each time after notifyTransferObject be notified or wait timeout for 1s. To reduce unnecessary block, at the end of loop, transferOk would be checked again to exit loop if tranfer success within timeout.

Reply:
I mean in line 285

Reply:
Got it. It may be better and more clear to check transferOk in the condition of while.